### ISSUE TITLE
FR: Print version of service while starting up

### ISSUE THREAD

**Issue by Dunemaster** [micronaut-projects/micronaut-core#9267]
### Feature description

As an application administrator, I want a quick and easy way to see the application version in the logs (after startup)

This may be useful for examing current state of a deployment or for posthumous investigation
________________________________________________________________________________________________________________
I am reading to implement this enhancement and I propose the following high-level design:

1) Define an interface for a version supplier bean

`
interface ApplicationVersionSupplier {
     @Nullable
     String version()
}
`

2) Create a default implementation, which will utilize JPMS module version
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/module/ModuleDescriptor.Version.html

other implementation may similar to gitInfoSource, build info source and son on
https://docs.micronaut.io/3.9.1/guide/#infoEndpoint

2.1) add an example of setting this version to micronaut docs/
See https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/compile/CompileOptions.html#getJavaModuleVersion--

3) Add an option to print this version after the banner
Will add a method similar to Micronaut::banner )https://docs.micronaut.io/3.9.1/api/index.html?io/micronaut/context/banner/MicronautBanner.html

`
@NonNull
public Micronaut version(VersionPrintPolicy printPolicy)
`

with VersionPrintPolicy having three options:

- Auto  (being the default) . Prints a version if it is available
- Never
- Always  (prints a warning about an empty version if no version can be supplied)

**Comment by graemerocher** [comment]
not sure why this proposal needs to rely on the module system

**Comment by Dunemaster** [comment]
@graemerocher, not necessarily. I propose JPMS to be one of several possible implementations, I can also inplement, for examle, another one using build-info.properties

**Comment by graemerocher** [comment]
Triage: this is a valid startup observability enhancement. Consider an opt-in version supplier SPI backed by build-info.properties, package metadata, or another application-provided source, then a banner/startup logging policy that prints the resolved version without requiring JPMS. Keep the default quiet or backward-compatible, define precedence and redaction rules, and add tests for absent, supplied, and multiple version sources.
